docs/explanation/development-process.rst

```diff
@@ -1,17 +1,320 @@
 
+Each release cycle follows the same general pattern, with the following
+major phases. Ubuntu contributors are expected to follow this process closely
+to ensure that their work is aligned with that of others. Because of the
+time-based release cycle, Ubuntu contributors must coordinate well to
+produce an on-time release.
+
+See also the article, :doc:`/explanation/releases`, for more details about the
+release cadence.
+
+Beginning a new release
+-----------------------
+
+The Ubuntu infrastructure is prepared for a new development branch at the
+beginning of each cycle. The package build system is set up, the toolchain
+is organised, :term:`seeds` are branched, and many other pieces are made ready
+before development can properly begin. Once these preparations are made, the
+new branch is officially announced on the
+`ubuntu-devel-announce mailing list <https://lists.ubuntu.com/mailman/listinfo/ubuntu-devel-announce>`_ 
+and opened for uploads to the :doc:`/explanation/archive`.
+
+.. note::
+    See the `Ubuntu 24.04 LTS (Noble Numbat) archive opening announcement email <https://lists.ubuntu.com/archives/ubuntu-devel-announce/2023-October/001341.html>`_
+    as an example.
+
+Planning
+--------
+
+Ubuntu contributors discuss the targeted features for each release cycle via
+the various channels (e.g., IRC, Matrix, Discourse, Launchpad). Some of these
+come from strategic priorities for the distribution as a whole, and some are
+proposed by individual developers.
+
+The broader open-source community gets together at the :term:`Ubuntu Summit` 
+(similar but different to the past 
+:term:`Ubuntu Developer Summits <Ubuntu Developer Summit>`) to share
+experiences and ideas and to inspire future exciting projects covering
+development as well as design, writing, and community leadership with a wide
+range of technical skill levels.
+
+The :term:`Canonical` teams meet after every release at the Roadmap Planning 
+Sprint to plan the roadmap for the next release and discuss how to implement
+the roadmap at the Engineering Sprint.
+
+Merging with Upstream & feature development
+-------------------------------------------
+
+The first phase of the release cycle is characterised by bringing new releases
+of :term:`upstream` components into Ubuntu, either directly or via 
+:doc:`Merges and Syncs from Debian </explanation/debian-merges-and-syncs>`. 
+Development of planned projects for the release often begins while merging is
+still underway, and their development is accelerated once the package archive
+is reasonably consistent and usable.
+
+The automatic import of new package versions from Debian ends at the 
+:ref:`DebianImportFreeze`.
+
+Stabilisation and deadlines (freezes)
+-------------------------------------
+
+During development, caution is increasingly exercised in making changes
+to Ubuntu to ensure a stable state is reached in time for the final release
+date. The typical order and length of the various freezes can be seen on the
```

Comment:
> It's not quite clear what the definition of a freeze is here. I think we could be a bit more explicit about what we mean by "freeze", to avoid using the term without making it really clear what the term means.

_Originally posted by @s-makin in https://github.com/canonical/ubuntu-packaging-guide/pull/54#discussion_r1624269513_